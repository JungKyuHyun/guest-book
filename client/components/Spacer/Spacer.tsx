import { css } from '@emotion/react';

type Space = 0 | 2 | 4 | 8 | 12 | 20 | 24 | 28 | 40;

type Props = {
  mt?: Space;
  mb?: Space;
};

export function Spacer({ mt = 0, mb = 0 }: Props) {
  return (
    <div
      css={css`
        display: block;
        margin-top: ${mt}px;
        margin-bottom: ${mb}px;
      `}
    />
  );
}
