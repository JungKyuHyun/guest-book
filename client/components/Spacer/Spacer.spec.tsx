import { cleanup } from '@testing-library/react';
import { matchers, createSerializer } from '@emotion/jest';
import renderer from 'react-test-renderer';
import { Spacer } from './Spacer';

expect.extend(matchers);
expect.addSnapshotSerializer(createSerializer());

describe('<Spacer />', () => {
  afterEach(cleanup);

  it('match snapshot', () => {
    const tree = renderer.create(<Spacer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render default style', () => {
    const tree = renderer.create(<Spacer />).toJSON();
    expect(tree).toHaveStyleRule('display', 'block');
    expect(tree).toHaveStyleRule('margin-top', '0px');
    expect(tree).toHaveStyleRule('margin-bottom', '0px');
  });

  it('should change margin-top style', () => {
    const tree = renderer.create(<Spacer mt={4} />).toJSON();
    expect(tree).toHaveStyleRule('margin-top', '4px');
  });

  it('should change margin-bottom style', () => {
    const tree = renderer.create(<Spacer mb={4} />).toJSON();
    expect(tree).toHaveStyleRule('margin-bottom', '4px');
  });

  it('should change margin-top and margin-bottom style', () => {
    const tree = renderer.create(<Spacer mt={4} mb={4} />).toJSON();
    expect(tree).toHaveStyleRule('margin-top', '4px');
    expect(tree).toHaveStyleRule('margin-bottom', '4px');
  });
});
