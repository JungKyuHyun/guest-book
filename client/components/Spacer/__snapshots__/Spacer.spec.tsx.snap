// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spacer /> match snapshot 1`] = `
.emotion-0 {
  display: block;
  margin-top: 0px;
  margin-bottom: 0px;
}

<div
  className="emotion-0"
/>
`;
